tests/models/xmod/test_modeling_xmod.py

```diff
@@ -0,0 +1,607 @@
+        self.assertTrue(torch.allclose(output[:, :, -1], expected_output_values_last_dim, atol=1e-3))
+
+        # language de_DE
+        model.set_default_language("de_DE")
+        input_ids = torch.tensor([[0, 1310, 49083, 443, 269, 71, 5486, 165, 60429, 660, 23, 2315, 58761, 18391, 5, 2]])
+        # Der Hund ist niedlich und wohnt in einem Gartenhaus.
+        expected_output_shape = torch.Size((1, 16, 768))  # batch_size, sequence_length, embedding_vector_dim
+        # fmt: off
+        expected_output_values_last_dim = torch.tensor(
+            [[0.0162, 0.0075, -0.1882, 0.2335, -0.0952, -0.3994, -0.0317, -0.1174, 0.0177, 0.4280, -0.0240, -0.2138,
+              0.0785, -0.1045, -0.2811, -0.3220]]
+        )
+        # fmt: on
+        output = model(input_ids)["last_hidden_state"].detach()
+        self.assertEqual(output.shape, expected_output_shape)
+        # compare the actual values for a slice of last dim
+        self.assertTrue(torch.allclose(output[:, :, -1], expected_output_values_last_dim, atol=1e-3))
+
+    @slow
+    def test_xmod_large_prenorm(self):
+        model = XmodModel.from_pretrained("jvamvas/xmod-large-prenorm")
+
+        # language en_XX
+        model.set_default_language("en_XX")
+        input_ids = torch.tensor([[0, 581, 10269, 83, 99942, 136, 60742, 23, 70, 80583, 18276, 2]])
+        # The dog is cute and lives in the garden house
+        expected_output_shape = torch.Size((1, 12, 1024))  # batch_size, sequence_length, embedding_vector_dim
+        # fmt: off
+        expected_output_values_last_dim = torch.tensor(
+            [[-0.0121, -0.0194, -0.0240, -0.0160, -0.0205, -0.0159, -0.0243, -0.0206, -0.0161, -0.0335, -0.0196,
+              -0.0141]]
+        )
+        # fmt: on
+        output = model(input_ids)["last_hidden_state"].detach()
+        self.assertEqual(output.shape, expected_output_shape)
+        # compare the actual values for a slice of last dim
+        self.assertTrue(torch.allclose(output[:, :, -1], expected_output_values_last_dim, atol=1e-3))
+
+        # language de_DE
+        model.set_default_language("de_DE")
+        input_ids = torch.tensor([[0, 1310, 49083, 443, 269, 71, 5486, 165, 60429, 660, 23, 2315, 58761, 18391, 5, 2]])
+        # Der Hund ist niedlich und wohnt in einem Gartenhaus.
+        expected_output_shape = torch.Size((1, 16, 1024))  # batch_size, sequence_length, embedding_vector_dim
+        # fmt: off
+        expected_output_values_last_dim = torch.tensor(
+            [[-0.0120, -0.0262, -0.0253, -0.0112, -0.0128, -0.0164, -0.0080, -0.0081, -0.0192, -0.0117, -0.0170,
+              -0.0120, -0.0210, -0.0173, -0.0078, -0.0122]]
+        )
+        # fmt: on
+        output = model(input_ids)["last_hidden_state"].detach()
+        self.assertEqual(output.shape, expected_output_shape)
+        # compare the actual values for a slice of last dim
+        self.assertTrue(torch.allclose(output[:, :, -1], expected_output_values_last_dim, atol=1e-3))
+
```

Comment:
A few tests are missing with example of `MaskedLM`, `XmodForCausalLM` etc. This could make sure that they also work. Very simple testing is enough, but would be great if all the `pipelines` work with the different variations of the model.

Reply:
This file already contains tests for `XmodForMaskedLM`, `XmodForCausalLM` etc.
The test coverage is the same as with other XLM-based models, e.g. `xlm_roberta_xl`.

Reply:
Thanks for clarifying.
I have now added an integration test for `FillMaskPipeline`.

I hesitate to check the output of other pipelines because there are no trained models for those pipelines. Checking the output of randomly initialized models amounts to a guarantee that different versions of `transformers` perform identical initialization. Such a guarantee would be out of the scope of this PR and should be discussed in a separate issue.

Reply:
It is perfect. If there are not pretrained model, it is not needed to test the pipelines !